homeassistant/components/zwave_js/__init__.py

```diff
@@ -1,31 +1,104 @@
 """The Z-Wave JS integration."""
 import asyncio
+import logging
+
+from async_timeout import timeout
+from zwave_js_server.client import Client as ZwaveClient
+from zwave_js_server.model.node import Node as ZwaveNode
 
 from homeassistant.config_entries import ConfigEntry
-from homeassistant.core import HomeAssistant
+from homeassistant.const import CONF_URL, EVENT_HOMEASSISTANT_STOP
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import device_registry
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
 
-from .const import DOMAIN
+from .const import DOMAIN, PLATFORMS
 
-PLATFORMS = ["sensor"]
+LOGGER = logging.getLogger(__name__)
 
 
 async def async_setup(hass: HomeAssistant, config: dict):
     """Set up the Z-Wave JS component."""
+    hass.data[DOMAIN] = {}
     return True
 
 
+@callback
+def register_node_in_dev_reg(
+    entry: ConfigEntry,
+    dev_reg: device_registry.DeviceRegistry,
+    client: ZwaveClient,
+    node: ZwaveNode,
+) -> None:
+    """Register node in dev reg."""
+    dev_reg.async_get_or_create(
+        config_entry_id=entry.entry_id,
+        identifiers={(DOMAIN, client.driver.controller.home_id, node.node_id)},
+        sw_version=node.firmware_version,
+        name=node.name,
+        model=node.product_type,
+        manufacturer=str(node.manufacturer_id),
+    )
+
+
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
-    """Set up Z-Wave JS from a config entry.
+    """Set up Z-Wave JS from a config entry."""
+    client = ZwaveClient(entry.data[CONF_URL], async_get_clientsession(hass))
+    initialized = asyncio.Event()
+
+    # pylint: disable=fixme
 
-    TODO Store an API object for your platforms to access
-    """
-    # hass.data[DOMAIN][entry.entry_id] = MyApi(...)
+    async def async_on_initialized():
+        """Handle initial full state received."""
+        # TODO: signal entities to update availability state
+        LOGGER.info("Connection to Zwave JS Server initialized")
+        initialized.set()
+
+        # This should be moved to the Node Ready command or something
+        # And then listen to new nodes too.
+        dev_reg = await device_registry.async_get_registry(hass)
+        for node in client.driver.controller.nodes.values():
+            register_node_in_dev_reg(entry, dev_reg, client, node)
+
+    async def async_on_disconnect():
+        """Handle websocket is disconnected."""
+        LOGGER.info("Disconnected from Zwave JS Server")
```

Comment:
Debug if it's expected. Otherwise warning.

Reply:
I think these are all still placeholders.